src/client/activation/requirements_txt/activator.ts

```diff
@@ -0,0 +1,32 @@
+import { injectable } from 'inversify';
+import { Hover, languages, TextDocument, Position } from 'vscode';
+import { IExtensionSingleActivationService } from '../types';
+
+@injectable()
+export class RequirementsTxtLinkActivator implements IExtensionSingleActivationService {
+    // eslint-disable-next-line class-methods-use-this
+    public async activate(): Promise<void> {
+        languages.registerHoverProvider(
+            {
+                pattern:
+                    '{**/requirements.txt,**/*-requirements.txt,**/requirements-*.txt,**/requirements.in,**/*-requirements.in,**/requirements-*.in,**/constraints.txt,**/*-constraints.txt,**/constraints-*.txt}',
+            },
+            {
+                provideHover(document: TextDocument, position: Position) {
+                    // source for Python distribution names valid characters: https://www.python.org/dev/peps/pep-0508/#names
+                    // Regex to match valid Python package name followed by semicolon, whitespace, or version specifiers
+                    // Matches package names up to 36 characters in length
+                    const packageRegex = /^([A-Z0-9]{0,36}|[A-Z0-9][A-Z0-9._-]{0,34}[A-Z0-9])(?=(;|\s|$|==|>|>=|<|<=|~=))/i;
```

Comment:
That lookahead assertion is going to be very costly. Is that there for line continuations, as that still not necessarily accurate as they can appear anywhere (see https://pip.pypa.io/en/latest/reference/requirements-file-format/#line-continuations).

Cheap way to do this is:

```suggestion
                    const packageRegex = /^([A-Z0-9]|[A-Z0-9][A-Z0-9._-]*[A-Z0-9])/i;
```

But unfortunately to be accurate you need to be looking _back_ to make sure that there wasn't a `\` on the previous line.

Reply:
What would be the best way to check for that? How many false positives would there be with the streamlined regex?

Reply:
I expect the simpler regex to handle most of the cases except where `\` is present between package names, which seems unlikely.